haystack/components/rankers/transformers_similarity.py

```diff
@@ -218,9 +218,15 @@ def run(
         :raises ValueError:
             If `top_k` is not > 0.
             If `scale_score` is True and `calibration_factor` is not provided.
-        :raises ComponentError:
+        :raises RuntimeError:
             If the model is not loaded because `warm_up()` was not called before.
         """
+        # If a model path is provided but the model isn't loaded
+        if self.model is None:
+            raise RuntimeError(
+                f"The component {self.__class__.__name__} wasn't warmed up. Run 'warm_up()' before calling 'run()'."
```

Comment:
```suggestion
                f"The component TransformersSimilarityRanker wasn't warmed up. Run 'warm_up()' before calling 'run()'."
```
Since the class name is not going to change.